xbmc/utils/Job.h

```diff
@@ -64,6 +65,19 @@ class IJobCallback
    \sa CJobManager and CJob
    */
   virtual void OnJobProgress(unsigned int jobID, unsigned int progress, unsigned int total, const CJob *job) {};
+
+  /*!
+   \brief An optional callback function used when a job will be aborted before it was executed.
+
+   OnJobAbort is called whenever a job gets aborted before it was executed. Job's DoWork method
+   will not be called, OnJobComplete will not be called.  The job instance will be destroyed right
```

Comment:
It looks to me that `OnJobAbort` directly calls `OnJobComplete`. Isn't that a contradiction to this documentation?

Reply:
No no, this is just a detail for CJobQueue implementing both call backs the same way, not  part of the general concept. CJobQueue has to do the same, regardless the job was executed or aborted before execution - namely to remove the job from its own processing queue and to continue processing by adding the next job(s) to the job manager. Other use cases might have very differently looking implementations for abort vs. completion of execution.

Reply:
Thank you for the clarification :+1: